solr/core/src/java/org/apache/solr/search/Grouping.java

```diff
@@ -461,7 +463,7 @@ private void searchWithTimeLimiter(final Query filterQuery, Collector collector)
       log.info("Query: {}; ", query);
       // to make WARN logged exception content more visible
       log.warn("Query: {}; ", query.getClass().getName(), x);
-      qr.setPartialResults(true);
+      qr.setPartialResults(shouldKeepPartials());
```

Comment:
Hmm, I'm not sure if this is correct - when these exceptions are thrown we know for sure that results are incomplete. Why would we set `partialResults=false` here? ? Unless there's an implicit behavior here that if we discard the results then we (always?) return `partialResults=false`.

Reply:
This seemed simplest and more readable. Code checks this value upstream in addition to it getting reported in the final output. Maybe if I track those down and ensure the up stream parsing handles it, this could be set to "omitted" that could remove the need for an additional response header value hmm.

Reply:
Next commit will contain the "omitted" value I mention, but discussion elsewhere indicates that even though I can maintain back compat it creates a tricky situation where the attribute is only sometimes boolean (and unquoted in the json) and other times can be a string (with quotes).  I'm probably better off just deprecating the partialResults flag and making a new string based attribute, perhaps "partialResultStatus" that can be "complete results", "partial results" or "results omitted" - that will come in the following commit